core/src/main/scala/org/http4s/Header.scala

```diff
@@ -141,56 +142,93 @@ object Header {
 
     /** Transform this header into a [[Header.Raw]]
       */
-    def toRaw(a: A): Header.Raw
+    def toRawOne(a: A): Header.Raw
+
+    /** Transform this (potentially repeating) header into a [[Header.Raw]] */
+    def toRaw(a: F[A]): Header.Raw
 
     /** Selects this header from a list of [[Header.Raw]]
       */
     def from(headers: List[Header.Raw]): Option[F[A]]
+
+    def fromSafe(headers: List[Header.Raw]): Option[Ior[NonEmptyList[ParseFailure], F[A]]]
```

Comment:
Trying to remember why we have an `Option[F]` on these, resulting in an `Option[NonEmptyList]` for `Recurring` instead of `F` being `Option` or `List`.

I don't love the name `fromSafe`, but I do think the `Ior` is a correct model: "give me all you can, and warn me about the ones you couldn't."

Reply:
> Trying to remember why we have an Option[F] on these, resulting in an Option[NonEmptyList] for Recurring instead of F being Option or List.

I could look into changing that, it makes more sense to me.

> I don't love the name fromSafe, but I do think the Ior is a correct model: "give me all you can, and warn me about the ones you couldn't."

I think just having 1 `from` that returns the `IOr` would be fine, and our users can `toOption` if they want to ignore the parse failures.

Reply:
@rossabaker thoughts?

Reply:
> I think just having 1 from that returns the IOr would be fine, and our users can toOption if they want to ignore the parse failures.
I think this makes sense.

Reply:
> I think just having 1 from that returns the IOr would be fine, and our users can toOption if they want to ignore the parse failures.

I agree with this.

Reply:
What about `Headers.get`? Should it return an `Option` or should it also return an `Option[Ior...`?

Reply:
Hmm.  Good question.  `My inclination is `Option[Ior[...`, but that would be a massive breaking change.  The existing signature is consistent with its map-like semantics.

I argued against the overload earlier, and I don't think the typeclass needs the overload, but `Headers` is the primary user API.  Maybe leaving `get` as is, and `getWithWarnings` that returns the `Option[Ior[...`?

Reply:
Done.